### PR TITLE
Don't overwrite adler32 hash with crc32 hash. 

### PR DESCRIPTION
**Pull request by nmoinvaz**
See #1066 for problem description.

### CODE REVIEW

**Comment by nmoinvaz** [comment]
I'm not sure if it should be `((state->wrap & 4) && (state->flags))`. I don't quite see exactly what the `state->wrap & 4` indicates.

File: test/inflate_adler_small.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include <stdlib.h>
+#include <string.h>
+
+z_const unsigned char compressed[] = {
+    0x78, 0x9c, 0x0b, 0xc9, 0x48, 0x55, 0x28, 0x2c, 0xcd, 0x4c, 0xce, 0x56, 0x48, 0x2a, 0xca,
+    0x2f, 0xcf, 0x53, 0x48, 0xcb, 0xaf, 0x50, 0xc8, 0x2a, 0xcd, 0x2d, 0x48, 0x4d, 0x51, 0xc8,
+    0x2f, 0x4b, 0x2d, 0x52, 0x28, 0xc9, 0x48, 0x55, 0xc8, 0x49, 0xac, 0xaa, 0x54, 0x48, 0xc9,
+    0x4f, 0x07, 0x00, 0x6b, 0x93, 0x10, 0x30
+};
+z_const unsigned char* uncompressed_expected = "The quick brown fox jumped over the lazy dog";
```
Comment (mtl1979) [inline test/inflate_adler_small.c]: Text literals are not "unsigned" on all compilers, so needs explicit cast.

**Comment by mtl1979** [comment]
@nmoinvaz `state->wrap & 4` should mean enable/disable check value... Basically in inflate, it is used to enable or disable crc32 calculation in raw mode.

**Comment by nmoinvaz** [comment]
@mtl1979 thanks for the great explanation. It is not documented very well so I have added a commit with some new comments.

**Comment by mtl1979** [comment]
@nmoinvaz Even I didn't know that before today... I had to grep through the sources to see what the purpose of that bit is...

**Comment by nmoinvaz** [comment]
Looks like CI passes now. This is good!

**Review by Dead2 — APPROVED** [review]

**Comment by nmoinvaz** [comment]
Thanks for merging so quickly!

**Pull request merged**